### PR TITLE
Switch to monaco editor

### LINKED ISSUE #150: move from ace editor to monaco editor
monaco: https://github.com/Microsoft/monaco-editor 

Few minimum requirements
 - [ ] should allow switching between light and dark theme. (monaco has support)
 - [ ] should provide _rich intellisense_ for Javascript

### PR DESCRIPTION
**Pull request by kapillamba4**
Fixes #150

### CODE REVIEW

File: src/components/editor/Shortcuts.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div id="shortcutId" class="btn-group">
     <button id="panelLang" type="button" class="btn btn-sm btn-menu" @click="showModal=true" @blur="showModal=false">
-      Shortcuts<i class="fa fa-reply-all" aria-hidden="true"></i>
+      Shortcuts&nbsp;<i class="fa fa-reply-all" aria-hidden="true"></i>
```
Comment (championswimmer) [inline src/components/editor/Shortcuts.vue]: use css to provide paddings, margins etc. &nbsp; usage is ugly :(

File: src/components/editor/Settings.vue [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
       <div class="btn-group">
-        <b>Font:</b>
+        <b>Font:</b>&nbsp;&nbsp;&nbsp;
         <select @change="changeFont">
           <option v-for="font in fontOptions" :value="font" :selected="setDefault('font',font)"> {{font}} </option>
         </select>
       </div>
       <div class="btn-group">
-        <b>Size:</b>
+        <b>Size:</b>&nbsp;&nbsp;&nbsp;
```
Comment (championswimmer) [inline src/components/editor/Settings.vue]: use css to provide paddings, margins etc. &nbsp; usage is ugly :(

**Comment by championswimmer** [comment]
Does it have rich intellisense for Javascript ?

**Comment by championswimmer** [comment]
apart from above comments, looks pretty good to me. great piece of work here !

**Comment by kapillamba4** [comment]
@championswimmer i have replaced ```&nbsp;``` by margin and I think it has [rich intellisense](https://microsoft.github.io/monaco-editor/#editor)

**Pull request merged**